README.md

```diff
@@ -78,103 +101,366 @@ Modify the app's **android/app/src/main/AndroidManifest.xml** to include the fol
+
+const options: RaygunClientOptions = {
+    apiKey: "This_Is_My_Key", 
+    version: "0.1.2-beta",
+    enableCrashReporting: true,
+    disableNativeCrashReporting: false,
+    enableRealUserMonitoring: true,
+    disableNetworkMonitoring: false,
+    customCrashReportingEndpoint: "https://myCrashReportingEndpoint.com",
+    customRealUserMonitoringEndpoint: "https://myRealUserMonitoringEndpoint.com",
+    onBeforeSendingCrashReport: (crashReport) => console.log(crashReport),
+    ignoredURLs: ["http://thisIsAnInternalCall.com", "http://thisIsAnotherInternalCall.com"]
+};
+
+// HERE
+RaygunClient.init(options);
 ```
 
-## addTag(... tags: string[])
-The `addTag` method adds tags to the Crash Reports and Real User Monitor. Within the Raygun application, 
+After successfully calling the init function, the client has been instantiated and initialized. 
+Now the other functions provided by the client will be able to run. 
+
+<br/>
+<br/>
+
+###  addTag(... tags: string[]) 
+The `addTag` function adds tags to the Crash Reports and Real User Monitor. Within the Raygun application, 
 you can use these tags to group particular errors together in a manner that aligns with what 
 you want.
+```typescript
+import RaygunClient from "raygun4reactnative"
 
-#### Example addTags
+RaygunClient.addTag("Invalid_User_Entry", "Caught_Exception");
+```
+
+<br/>
+<br/>
 
+###  setUser(user: User | string) 
+The `setUser` function get parse a [User](#user) object, or a string (used as an ID). Updating the user
+will aid in Real User and Network monitoring session data. To reset the user back to ANON, simply call
+this method with `null` as the 'user' parameter.
 ```typescript
-RaygunClient.addTag("Invalid_User_Entry", "Caught_Exception");
+import RaygunClient, {User} from "raygun4reactnative"
+
+const user: User = {
+    identifier: 'identifier',
+    isAnonymous: false,
+    email: 'user_by_object@email.com',
+    firstName: 'first name',
+    fullName: 'full name',
+    uuid: 'uuid'
+}
+
+RaygunClient.setUser(user);
+
+// Alternatively
+RaygunClient.setUser("identifier");
+
+// Reset to ANON user
+RaygunClient.setUser(null);
+```
+
+<br/>
+<br/>
+
+###  clearSession() 
+Clears all information about the user and tags that have been set to this point.
+```typescript
+import RaygunClient from "raygun4reactnative"
+
+RaygunClient.clearSession();
+```
+
+<br/>
+<br/>
+
+###  recordBreadcrumb(message: string, details?: BreadcrumbOption) 
+Records Breadcrumbs that will be sent away with the CrashReportPayload.
+```typescript
+import RaygunClient, {BreadcrumbOption} from "raygun4reactnative"
+
+const bco: BreadcrumbOption = {
+    category: 'Some String you choose',
+    level: 'debug',
+    customData: {"Key_6": "My Data is bland"},
+}
+
+RaygunClient.recordBreadcrumb("Some message", bco);
+```
+
+<br/>
+<br/>
+
+###  sendError(error: Error, ...params: any) 
+Allows for an error to be sent to the Crash Reporting error handler along with some customized
```

Comment:
Sentence could be simplified to "Manually sends an error to Raygun. This method supports adding additional tags or custom data which will be appended to any other tags or custom data already provided."